src/components/userService.ts

```diff
@@ -0,0 +1,47 @@
+import { Memento } from 'vscode';
+
+export class GlobalStateTokenStorage {
+	constructor(private readonly __globalState: Memento) {}
+
+	getUserToken() {
+		const clerkToken = this.__globalState.get('userToken');
+		return typeof clerkToken === 'string' ? clerkToken : null;
+	}
+
+	setUserToken(clerkToken: string | undefined) {
+		this.__globalState.update('userToken', clerkToken);
+	}
+}
+
+export class AlreadyLinkedError extends Error {
+	constructor() {
+		super('Already linked to different account');
+	}
+}
+
+export interface IGlobalStateTokenStorage {
+	getUserToken(): string | null;
+	setUserToken(value: string | undefined): void;
+}
+
+export class UserService {
+	constructor(private readonly __storage: IGlobalStateTokenStorage) {}
```

Comment:
you don't need to use the dependency injection interface pattern here, unless there's a reason for that?

Reply:
addressed